Helper/Api/Products.php

```diff
@@ -1130,19 +1137,29 @@ private function processImages(array $flat_product, Product $target, bool $shoul
                     }
                     $shouldUpdate = $this->setGalleryImage($product_image['big_url'], $target, $mainImage);
```

Comment:
```suggestion
                    $shouldUpdate = $this->setGalleryImage($product_image['big_url'], $target, $mainImage) || $shouldUpdate ;
```

otherwise you overwrite the update need for product removal